File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
    result = []
    for m in multiples:
        [
            result.append(n)
            for n in range(limit)
            if m != 0 and n % m == 0
        ]
    return sum(set(result))


if __name__ == "__main__":
    print(f"{sum_of_multiples(10000, [43, 47])=}")

File: go/sum-of-multiples/sum_of_multiples.go
// Package summultiples implements a solution for the exercism sum-of-multiples chalenge
package summultiples

// SumMultiples gives the sum of all the numbers up to limit that are multiples of any of the divisors
func SumMultiples(limit int, divisors ...int) int {
	result := 0

	for n := 1; n < limit; n++ {
		for _, d := range divisors {
			if d == 0 {
				continue
			}
			if n%d == 0 {
				result += n
				break
			}
		}
	}

	return result
}
